sympy/core/expr.py

```diff
@@ -3491,7 +3519,7 @@ def round(self, n=None):
             if n is None:  # the single-arg case
                 return rv
             # use str or else it won't be a float
-            return Float(str(rv), digits_needed)
+            return Float(str(rv), dps)  # keep same precision
```

Comment:
This lets `round(12345.0, -4)` be `Float(10000.0)` instead of `Float(10000, 2)`.